app/controllers/kaui/admin_tenants_controller.rb

```diff
@@ -153,6 +153,27 @@ def delete_catalog
 
   def new_plan_currency
     @tenant = safely_find_tenant_by_id(params[:id])
+
+    is_plan_id_found = false
+    plan_id = params[:plan_id]
+
+    options = tenant_options_for_client
+    options[:api_key] = @tenant.api_key
+    options[:api_secret] = @tenant.api_secret
+
+    catalogs = Kaui::Catalog::get_catalog_json(false, options)
+
+    # seek if plan id exists
+    catalogs.each do |catalog|
```

Comment:
I think We only need to look at the latest version. If a plan is not in the latest version, it means it has been decommissioned, so i would not expect to allow admin to add new currencies anyway.